### PR TITLE
Feature parse

### PR DESCRIPTION
**Pull request by dwsideriusNIST**
Proposal for new parser for isotherm_data block:

1. Converts various delimiters to whitespace (tabs are column delimiters in paste-from-excel, at least on my Mac)
1. Collapses whitespace in case multiple spaces are between columns
1. Uses Pandas to interpret the column data, then sends the output to ndarray

One outstanding issue: the separator declaration is ',| ' - but the comma should be unnecessary. When the comma is not included, pandas creates too many columns when lines have _trailing_ whitespace. I don't yet understand this yet, need to consult docs

Will include examples below

### CODE REVIEW

**Comment by dwsideriusNIST** [comment]
Example of pastes from non-CSV data

Example 1 (paste from MS Excel):

0.03362 | 0.1127
0.08609 | 0.3133
0.15333 | 0.5590
0.46870 | 1.5469
0.72747 | 2.1537
0.92713 | 2.5158
1.07506 | 2.7393
1.21081 | 2.9156
1.40979 | 3.132

That didn't work, as the tab was converted to |. Please try a direct paste from MS Excel if you have it. I've also tested this with LibreOffice Calc, Google Sheets, and Gnumeric, all of which worked fine.

Example 2: using too much white space. (Libreoffice does this.)

0.03362 0.1127
0.08609 0.3133
0.15333 0.5590
0.46870 1.5469
0.72747 2.1537

Example 3:

0.03362 ; 0.1127
0.08609 ; 0.3133
0.15333 ; 0.5590
0.46870 ; 1.5469
0.72747 ; 2.1537

File: digitizer/parse.py [hunk 1]
```diff
@@ -52,12 +53,22 @@ def prepare_isotherm_dict(form):
...
-    measurements = np.genfromtxt(StringIO(measurements),
-                                 delimiter=',',
-                                 comments='#')
-    measurements = np.array(measurements, ndmin=2)  # deal with single data row
+    measurements = form.inp_isotherm_data.value  # get string from input text box
+    for delimiter in ['\t', ';', '|', ',']:
+        measurements = measurements.replace(
+            delimiter, ' ')  # convert all delimiters to spaces
+    measurements = re.sub(' +', ' ', measurements)  # collapse whitespace
+    print(measurements)
```
Comment (ltalirz) [inline digitizer/parse.py]: leftover print statement

File: digitizer/parse.py [hunk 2]
```diff
@@ -52,12 +53,22 @@ def prepare_isotherm_dict(form):
...
+    measurements = pd.read_table(
+        StringIO(measurements),
+        sep=',| ',
+        #sep=' ',
+        # for some reason, leaving only the space delimiter is causing a problem
+        #  when lines have trailing whitespace. Need to check pandas documentation
```
Comment (ltalirz) [inline digitizer/parse.py:65]: I see. Alternatively, we could iterate over the lines and `strip` them

File: digitizer/parse.py [hunk 3]
```diff
@@ -2,7 +2,8 @@
 """Prepare JSON output."""
 from io import StringIO
 import re
-import numpy as np
+# import numpy as np
```
Comment (ltalirz) [inline digitizer/parse.py]: can be removed

**Review by ltalirz — CHANGES_REQUESTED** [review]
thanks, dan!

some very minor requests; after that we can already merge it if you like

File: digitizer/parse.py (hunk 2 above)
Comment (dwsideriusNIST) [reply digitizer/parse.py:65]: That was the inelegant solution that came to mind to me as well. Will still check pandas docs

File: digitizer/parse.py (hunk 1 above)
Comment (dwsideriusNIST) [reply digitizer/parse.py]: fixed

File: digitizer/parse.py (hunk 3 above)
Comment (dwsideriusNIST) [reply digitizer/parse.py]: done

**Pull request merged**